design/open/kv-query-tool.md

```diff
@@ -0,0 +1,45 @@
+# KV Store Querying Tool
+
+## Scope
+
+Our newly introduced KV Store stores the data in the form of [`Partition Key`, `Key`, `Value`] triplets. The `partition key` and `key` fields are strings stored as byte arrays and the `value` is an object (various types) serialized as protobuf and stored as byte array.
```

Comment:
Key is byte slice, not string

Reply:
I think `partition_key` too

Reply:
We can always represent `partition_key` as a string, we can't give that guarantee for `key` (e.g. staging entry)

Reply:
That's right for the keys (partition and key) that `lakefs` send to the KV store. But somewher in the process, lakefs translates strings into byte slices. I want to use the same logic for the querying tool.
Eventually, a `CLI`-ish tool, will have to use readable strings, at least at the `I` part. WDYT?

Reply:
Keep in mind that `keys` in contrast to `partition_keys` are by design (graveler constraint) byte slices - we can't assume they have any encoding or have a valid string representation

Reply:
@N-o-Z - I believe you are specifically refering to the staging object path, that may contain an unreadable chars.
In that case, or any other case we cannot guarantee displayable encoding, I cannot see how can we access/display these from a human-usable, string based, CLI tool
We might want to skip the `key` part here, and only allow scan. This will display some kind of key - not neccessarily readable, and the formatted data
This will require the tool to specifically handle known caveats, such as this, and is not generally scalable, but apparently we will have to address it

Reply:
Modified doc to reflect the challenge presented by non printable keys, and offer optional bypassing